### PR TITLE
SQL Server Certification Tests

### PR DESCRIPTION
**Pull request by berndverst**
# Description

SQL Server certification tests as per #949 

## Issue reference

#949

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/certification/state/sqlserver/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+This project aims to test the SQL Server State Store component under various conditions.
+
+## Test plan
+
+### SQL Injection
+
+* Not prone to SQL injection on write
+* Not prone to SQL injection on read
+* Bit oribe ti SQK injection on delete
```
Comment (CodeMonkeyLeet) [inline tests/certification/state/sqlserver/README.md:11]:
```suggestion
* Not prone to SQL injection on delete
```

File: tests/certification/state/sqlserver/go.mod [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
+	github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d // indirect
+	github.com/andybalholm/brotli v1.0.2 // indirect
+	github.com/antlr/antlr4 v0.0.0-20200503195918-621b933c7a7f // indirect
+	github.com/armon/go-metrics v0.0.0-20190430140413-ec5e00d3c878 // indirect
+	github.com/beorn7/perks v1.0.1 // indirect
+	github.com/cenkalti/backoff v2.2.1+incompatible // indirect
+	github.com/cenkalti/backoff/v4 v4.1.1 // indirect
+	github.com/cespare/xxhash/v2 v2.1.1 // indirect
+	github.com/dapr/go-sdk v1.2.1-0.20211017032306-de68193d5cd9 // indirect
```
Comment (CodeMonkeyLeet) [inline tests/certification/state/sqlserver/go.mod:29]: `go mod tidy` this, it looks like you have a direct reference on line 40 of the test.

Side note: There's an issue with /ok-to-test where the deduplication task in the existing workflows ignores test changes, so it won't run this.

**Review by artursouza — APPROVED** [review]

**Pull request merged**